### PR TITLE
Wait as default

### PR DESCRIPTION
**Pull request by dylan-smith**
- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created) - #946 
- [x] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

Implements #705 

Changing the default behavior of `migrate-repo` and `migrate-org` to be the same as the `--wait` flag.

The `--wait` flag is now unnecessary, but for compat reasons I'm leaving it in. I made it a hidden option, and if you pass it you'll get this warning: `--wait flag is obsolete and will be removed in a future version. The default behavior is now to wait.`

Added a new option `--queue-only` which will result in the same behavior as the previous default behavior.

Also updating `generate-script` commands to include the `--queue-only` flag for parallel scripts (and remove `--wait` from the sequential scripts).

For customers that have pre-existing parallel scripts (without the `--wait` flag) this will result in a change in behavior (scripts that used to be parallel will now run sequentially), but it shouldn't **break** their scripts (the integration tests confirm this). Anybody that relies on the default behavior will start getting a warning that says:
`[WARNING] The default behavior has changed from only queueing the migration, to waiting for the migration to finish. If you ran this as part of a script to run multiple migrations in parallel, consider using the new --queue-only option to preserve the previous default behavior.`. In a future release we'll remove this warning.

Any existing sequential scripts will continue to work just fine, they will just see warnings about using an unnecessary `--wait` flag now.

Added validation that both `--wait` and `--queue-only` can't be passed in together.

### CODE REVIEW

**Comment by dylan-smith** [comment]
> There definitely are breaking change concerns with this update, but I still think we should make the jump and fix this behaviour - especially before GA.
> 
> At worst, it'll be a minor annoyance if your script which used to run now waits, and you can change it!
> 
> What do you think of adding a temporary warning for a version or two to give users a heads-up about the change in behaviour?

Just to make sure I understand your suggestion, you are suggesting we first roll out a change that adds the `--queue-only` flag, but leaves the default behavior the same as today, and spits out a warning if you don't pass either `--wait` or `--queue-only` saying something like `the default behavior will change in the future, you should pass --queue-only flag instead of relying on default behavior`. Then a few weeks down the road we roll out the change to the default behavior. Then a couple months after that we remove the `--wait` flag entirely.

Because I considered doing it that way, but the fact that we are not actually breaking anybodies scripts, we're just making them run slower made me lean towards just pulling the bandaid off all at once.

But I'll defer to your opinion on this one.

**Comment by timrogers** [comment]
I was thinking that we change the default immediately, but add a temporary log line as a heads up about the change.

**Comment by dylan-smith** [comment]
> I was thinking that we change the default immediately, but add a temporary log line as a heads up about the change.

Oh I see, so maybe a warning is somebody runs it without either `--wait` or `--queue-only` to let them know the behavior has changed.

**Comment by timrogers** [comment]
Exactly. That's what I was thinking.

**Review by ArinGhazarian — APPROVED** [review]

File: src/OctoshiftCLI.Tests/bbs2gh/Handlers/GenerateScriptCommandHandlerTests.cs [hunk 1]
```diff
@@ -476,7 +476,7 @@ public async Task One_Repo_With_No_Ssl_Verify()
             SshPort = SSH_PORT,
             Output = new FileInfo(OUTPUT),
             Verbose = true,
-            NoSslVerify = true
+            NoSslVerify = true,
```
[the remark this answers is not in the document]
Comment (dylan-smith) [reply src/OctoshiftCLI.Tests/bbs2gh/Handlers/GenerateScriptCommandHandlerTests.cs:479]: that's on purpose, and generally considered a good practice. Makes it easier to add new lines/properties after it, and the subsequent diff will be cleaner.

**Pull request merged**